src/PSR7/OperationAddress.php

```diff
@@ -4,24 +4,103 @@
 
 namespace OpenAPIValidation\PSR7;
 
-class OperationAddress extends PathAddress
+use OpenAPIValidation\PSR7\Exception\Validation\InvalidPath;
+use function is_numeric;
+use function preg_match;
+use function preg_match_all;
+use function preg_replace;
+use function sprintf;
+use function str_replace;
+use function strtok;
+
+class OperationAddress
 {
     /** @var string */
     protected $method;
+    /** @var string */
+    protected $path;
 
     public function __construct(string $path, string $method)
     {
-        parent::__construct($path);
+        $this->path   = $path;
         $this->method = $method;
     }
 
+    /**
+     * Checks if path matches a specification
+     *
+     * @param string $specPath like "/users/{id}"
+     * @param string $path     like "/users/12"
+     */
+    public static function isPathMatchesSpec(string $specPath, string $path) : bool
+    {
+        $pattern = '#^' . preg_replace('#{[^}]+}#', '[^/]+', $specPath) . '/?$#';
+
+        return (bool) preg_match($pattern, $path);
+    }
+
     public function method() : string
     {
         return $this->method;
     }
 
-    public function getOperationAddress() : self
```

Comment:
I remove casting method, since there is no sence in it. Class is immutable VO and successors can use it signature without casting